File: src/main/java/com/paragon/client/systems/module/impl/render/MobOwner.java
package com.paragon.client.systems.module.impl.render;

import com.paragon.api.util.entity.EntityUtil;
import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.module.Category;
import com.paragon.api.module.Module;
import net.minecraft.entity.IEntityOwnable;
import net.minecraft.entity.passive.AbstractHorse;
import net.minecraft.util.math.Vec3d;

import java.util.Objects;

public class MobOwner extends Module {

    public MobOwner() {
        super("MobOwner", Category.RENDER, "Displays the owner of a mob");
    }

    @Override
    public void onRender3D() {
        mc.world.loadedEntityList.forEach(entity -> {
            if (entity instanceof IEntityOwnable) {
                String name = entity.getCustomNameTag().equals("") ? entity.getName() : entity.getCustomNameTag();
                String owner = ((IEntityOwnable) entity).getOwner() == null ? "No Owner" : Objects.requireNonNull(((IEntityOwnable) entity).getOwner()).getName();
                Vec3d vec = EntityUtil.getInterpolatedPosition(entity);

                if (entity instanceof AbstractHorse) {
                    RenderUtil.drawNametagText("Name: " + name + ", Speed: " + ((AbstractHorse) entity).getAIMoveSpeed() + ", Owner: " + owner, new Vec3d(vec.x, vec.y + 1.25, vec.z), -1);
                } else {
                    RenderUtil.drawNametagText("Name: " + name + ", Owner: " + owner, new Vec3d(vec.x, vec.y + 1.25, vec.z), -1);
                }
            }
        });
    }
}

File: src/main/kotlin/com/paragon/client/systems/module/impl/render/MobOwner.kt
package com.paragon.client.systems.module.impl.render

import com.paragon.api.module.Category
import com.paragon.api.module.Module
import com.paragon.api.util.entity.EntityUtil
import com.paragon.api.util.render.RenderUtil.drawNametagText
import net.minecraft.entity.IEntityOwnable
import net.minecraft.entity.passive.AbstractHorse
import net.minecraft.util.math.Vec3d

object MobOwner : Module("MobOwner", Category.RENDER, "Displays the owner of a mob") {

    override fun onRender3D() {
        minecraft.world.loadedEntityList.forEach {
            if (it is IEntityOwnable) {
                val name = if (it.customNameTag == "") it.name else it.customNameTag
                val owner = if ((it as IEntityOwnable).owner == null) {
                    "No Owner"
                } else (it as IEntityOwnable).owner?.name

                val vec = EntityUtil.getInterpolatedPosition(it)

                if (it is AbstractHorse) {
                    drawNametagText(
                        "Name: " + name + ", Speed: " + (it as AbstractHorse).aiMoveSpeed + ", Owner: " + owner,
                        Vec3d(vec.x, vec.y + 1.25, vec.z),
                        -1
                    )
                } else {
                    drawNametagText("Name: $name, Owner: $owner", Vec3d(vec.x, vec.y + 1.25, vec.z), -1)
                }
            }
        }
    }

}